src/App.tsx

```diff
@@ -15,71 +17,93 @@ export const goods = [
+  selectedGood: string,
+};
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  emptyState = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {
+          selectedGood
+            ? (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={
+                    this.emptyState
+                  }
+                />
+              </h1>
+            )
+            : (
+              <h1 className="title">
+                No goods selected
+              </h1>
+            )
+        }
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              return (
+                <tr
+                  data-cy="Good"
+                  className={classNames({
+                    'has-background-success-light': selectedGood === good,
+                  })}
+                >
+                  <td>
+                    <button
+                      data-cy={
+                        selectedGood === good
+                          ? 'RemoveButton'
+                          : 'AddButton'
+                      }
+                      type="button"
+                      className={classNames(
+                        'button',
+                        { 'is-info': selectedGood === good },
+                      )}
+                      onClick={() => {
+                        if (selectedGood === good) {
+                          this.emptyState();
+                        } else {
+                          this.setState({ selectedGood: good });
+                        }
+                      }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)